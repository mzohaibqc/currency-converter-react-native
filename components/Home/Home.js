import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

import Screen from 'components/Screen';
import AppLogo from 'components/AppLogo';
import CurrencyInput from 'components/CurrencyInput';
import InputsContainer from 'components/InputsContainer';
import { SCREENS } from 'constants';
import useScreenTransform from 'hooks/useScreenTransform';
import SwitchIcon from 'assets/transfer.svg';
import selectors from 'selectors';
import {
  switchCurrencies,
  setBaseCurrencyValue,
  setTargetCurrencyValue,
} from 'actions/currency.actions';

function Home({ navigation }) {
  const theme = useSelector(selectors.getTheme);
  const baseCurrency = useSelector(selectors.getBaseCurrency);
  const targetCurrency = useSelector(selectors.getTargetCurrency);
  const baseCurrencyValue = useSelector(selectors.getBaseCurrencyValue);
  const exchangeRate = useSelector(selectors.getExchangeRate);
  const targetCurrencyValue = useSelector(selectors.getTargetCurrencyValue);
  const dispatch = useDispatch();

  const { imageY, inputContainerY, imageScale } = useScreenTransform();

  return (
    <Screen backgroundColor={theme.color}>
      <AppLogo scale={imageScale} translateY={imageY} showInfo />
      <InputsContainer translateY={inputContainerY}>
        <Title testID="main-title">Currency Converter</Title>
        <CurrencyInput
          code={baseCurrency}
          changeCurrency={(item) => {
            navigation.navigate(SCREENS.CURRENCIES, {
              title: 'Base Currency',
            });
          }}
          onChange={(value) => {
            dispatch(setBaseCurrencyValue(value));
          }}
          value={baseCurrencyValue}
        />
        <RowFlex justify="flex-start">
          <SwitchIconContainer onPress={() => dispatch(switchCurrencies.pending())} >
            <SwitchIcon height={40} width={30}  fill="#fff" />
          </SwitchIconContainer>
        </RowFlex>

        <CurrencyInput
          code={targetCurrency}
          changeCurrency={(item) => {
            navigation.navigate(SCREENS.CURRENCIES, {
              title: 'Target Currency',
            });
          }}
          onChange={(value) => {
            dispatch(setTargetCurrencyValue(value));
          }}
          value={targetCurrencyValue}
        />
        <RowFlex marginTop={15}>
          {exchangeRate && (
            <InfoText>{`1 ${baseCurrency} = ${Number(exchangeRate).toFixed(
              3,
            )} ${targetCurrency} as of ${new Date()
              .toDateString()
              .slice(4, 10)}`}</InfoText>
          )}
        </RowFlex>
      </InputsContainer>
    </Screen>
  );
}

Home.propTypes = {
  navigation: PropTypes.object,
};

export default Home;

const Title = styled.Text`
  color: #fff;
  font-size: 26px;
  text-align: center;
  margin-bottom: 10px;
`;

const RowFlex = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: ${(props) => props.justify || 'center'};
  margin-top: ${(props) => `${props.marginTop || 0}px`};
`;

const InfoText = styled.Text`
  color: #fff;
  font-size: 16px;
  font-weight: 500;
`;
const SwitchIconContainer = styled.TouchableOpacity`
  width: 30px;
  height: 40px;
  padding-left: 10px;
`;
