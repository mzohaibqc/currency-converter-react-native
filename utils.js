export const getAsyncActionType = (name) => {
  return {
    PENDING: `${name}: PENDING`,
    SUCCESS: `${name}: SUCCESS`,
    ERROR: `${name}: ERROR`,
    toString: () => name,
  };
};

export const sleep = n => new Promise(res => setTimeout(res, n));
