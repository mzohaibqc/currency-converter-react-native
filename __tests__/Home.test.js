import React from 'react';
import { render, cleanup, fireEvent } from 'react-native-testing-library';
import 'jest-styled-components';
import { customRender } from './utils';

import Home from 'components/Home/Home';

afterEach(cleanup);



describe('<Home />', () => {  
  it('should have title text', () => {
    const { getByText, baseElement } = customRender(
      <Home />
    );
    expect(getByText('Currency Converter')).toBeTruthy()
    expect(baseElement).toMatchSnapshot()
  });
});

// test('renders correctly', () => {
//   const tree = renderer.create(<Home />).toJSON();
//   expect(tree).toMatchSnapshot();
// });


// test('examples of some things', async () => {
//   const { getByTestId, getByText, queryByTestId, baseElement } = render(<Home />);
//   // const famousWomanInHistory = 'Ada Lovelace';

//   // const input = getByTestId('input');
//   // fireEvent.changeText(input, famousWomanInHistory);

//   // const button = getByText('Print Username');
//   // fireEvent.press(button);

//   // await wait(() => expect(queryByTestId('printed-username')).toBeTruthy());

//   // expect(getByTestId('printed-username').props.children).toBe(famousWomanInHistory);
//   // expect(baseElement).toMatchSnapshot();
// });