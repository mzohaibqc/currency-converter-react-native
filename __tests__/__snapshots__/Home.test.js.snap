// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should have title text 1`] = `undefined`;
